### PR TITLE
Make development-environment platform-independent

### PR DESCRIPTION
**Pull request by manuth**
This PR will make the `copy-metafiles` script alongside with some other scripts platform independent by using `node` rather than a command.
This script could be exported to an actual `.js`-file and invoked using `node ./copyMetafiles.js`.

Related to #1781

### CODE REVIEW

**Comment by ljharb** [comment]
Oops, turns out `fs.copyFileSync` doesn't exist on node <= 6.

**Comment by manuth** [comment]
Personally I wouldn't care that much about node <= 6 as the current LTS version is 12
I don't think there are any OSes and PCs out there you want to develop nodejs on which natively install nodejs v6.

But if not dropping there seems to be a polyfill-module on npm for exactly this purpose:
https://www.npmjs.com/package/fs-copy-file-sync

**Comment by ljharb** [comment]
If we need to use a dep, then we might as well move it to a script file.

**Comment by manuth** [comment]
@ljharb alright
I created separate scripts for all non-cross platform commands.
Please keep in mind that windows-tests still will fail as `windows-fix` isn't merged yet.

File: package.json [hunk 1]
```diff
@@ -21,15 +21,15 @@
     "prebuild": "rimraf lib",
     "build": "babel --quiet --out-dir lib src",
     "postbuild": "npm run copy-metafiles",
-    "copy-metafiles": "for DIR in memo-parser resolvers/node resolvers/webpack utils; do cp LICENSE .npmrc \"${DIR}/\"; done",
+    "copy-metafiles": "node -e \"require('babel-register'); require('./scripts/copyMetafiles');\"",
```
Comment (ljharb) [inline package.json:24]:
```suggestion
    "copy-metafiles": "node --require=babel-register ./scripts/copyMetafiles",
```

File: package.json [hunk 2]
```diff
@@ -21,15 +21,15 @@
...
     "watch": "npm run tests-only -- -- --watch",
     "pretest": "linklocal",
     "posttest": "eslint .",
     "mocha": "cross-env BABEL_ENV=test nyc -s mocha",
     "tests-only": "npm run mocha tests/src",
     "test": "npm run tests-only",
     "test-compiled": "npm run prepublish && BABEL_ENV=testCompiled mocha --compilers js:babel-register tests/src",
-    "test-all": "npm test && for resolver in ./resolvers/*; do cd $resolver && npm test && cd ../..; done",
+    "test-all": "node -e \"require('babel-register'); require('./scripts/testAll');\"",
```
Comment (ljharb) [inline package.json:32]:
```suggestion
    "test-all": "node --require=babel-register ./scripts/testAll",
```

File: scripts/resolverDirectories.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import glob from 'glob'
+
+export default glob.sync('./resolvers/*/')
```
Comment (ljharb) [inline scripts/resolverDirectories.js:3]: trailing newline

File: scripts/testAll.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+let npmPath = npmWhich(__dirname).sync('npm')
+let spawnOptions = {
+    stdio: [process.stdin, process.stdout, process.stderr],
+}
+
+spawnSync(
+    npmPath,
+    ['test'],
+    Object.assign({ cwd: __dirname }, spawnOptions))
```
Comment (ljharb) [inline scripts/testAll.js:13]:
```suggestion
    { cwd: __dirname ...spawnOptions }
```
since we're using babel anyways

File: scripts/testAll.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+for (let resolverDir of resolverDirectories) {
+    spawnSync(
+        npmPath,
+        ['test'],
+        Object.assign({ cwd: resolverDir }, spawnOptions))
```
Comment (ljharb) [inline scripts/testAll.js:19]:
```suggestion
        { cwd: resolverDir, ...spawnOptions }
```

File: scripts/testAll.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { spawnSync } from 'child_process'
+import npmWhich from 'npm-which'
+import resolverDirectories from './resolverDirectories'
+
+let npmPath = npmWhich(__dirname).sync('npm')
+let spawnOptions = {
+    stdio: [process.stdin, process.stdout, process.stderr],
```
Comment (ljharb) [inline scripts/testAll.js:7]:
```suggestion
    stdio: 'inherit',
```
i think?

File: package.json (hunk 1 above)
Comment (manuth) [reply package.json:24]: Oh - awesome, didn't know this is possible 😅

**Comment by manuth** [comment]
@ljharb object spreading doesn't seem to be supported by this babel-setup

**Comment by ljharb** [comment]
Had to tweak the CI script; the metafiles script was previously ran prior to running `npm install`.

**Pull request merged**